File: src/main/java/com/ottofernan/librarycrud/domain/models/Author.java
package com.ottofernan.librarycrud.domain.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.Entity;
import javax.persistence.ManyToMany;
import javax.persistence.Table;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "Authors")
public class Author extends Person{

    @JsonIgnore
    @ManyToMany(mappedBy = "authors")
    private Set<Book> books = new HashSet<>();

    public Set<Book> getBooks() {
        return books;
    }

    public void setBooks(Set<Book> books) {
        this.books = books;
    }

    @Override
    public String toString() {

        return getFirstName() + " " + getLastName();
    }
}

File: src/main/java/com/ottofernan/librarycrud/domain/models/Author.kt
package com.ottofernan.librarycrud.domain.models

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.Entity
import javax.persistence.ManyToMany
import javax.persistence.Table

@Entity
@Table(name = "Authors")
class Author: Person() {

    @JsonIgnore
    @ManyToMany(mappedBy = "authors")
    var books = mutableSetOf<Book>(Book())

    override fun toString(): String =
            "$firstName $lastName"
}